### PR TITLE
Edit minute

### PR DESCRIPTION
**Pull request by moiseslorap**
added edit minute and unit tests

### CODE REVIEW

File: app/minutes/controllers.py [hunk 1]
```diff
@@ -303,6 +303,39 @@ def delete_topics(user, user_data):
...
+
+    if minute is None:
+        emit('edit_minute', Response.MinuteDoesntExist)
+        return
+    
+    committee = minute.committee 
+
+    membership = committee.members.filter_by(member = user).first()
+
+    if minute.private and (membership is None and not user.is_admin and committee.head != user.id):
```
Comment (omardlhz) [inline app/minutes/controllers.py]: This should be changed to:
```python
if (user.id != committee.head and user.is_admin == False and
        (membership is None or membership.role != Roles.MinuteTaker)):
```

File: app/minutes/controllers.py [hunk 2]
```diff
@@ -303,6 +303,39 @@ def delete_topics(user, user_data):
...
+@ensure_dict
+@get_user
+def edit_minute(user, user_data):
+    minute = Minutes.query.filter_by(id= user_data.get("minute_id", -1)).first()
+
+    if user is None:
+        emit('edit_minute', Response.UserDoesntExist)
+        return
+
+    if minute is None:
```
Comment (omardlhz) [inline app/minutes/controllers.py]: Please test case in which minute doesn't exist.

File: app/minutes/controllers.py [hunk 3]
```diff
@@ -303,6 +303,39 @@ def delete_topics(user, user_data):
...
         emit("delete_minute_topics", Response.DeleteTopicError)
 
+@socketio.on('edit_minute')
+@ensure_dict
+@get_user
+def edit_minute(user, user_data):
+    minute = Minutes.query.filter_by(id= user_data.get("minute_id", -1)).first()
+
+    if user is None:
+        emit('edit_minute', Response.UserDoesntExist)
```
Comment (omardlhz) [inline app/minutes/controllers.py]: Test that user doesn't exist.

File: app/minutes/controllers.py [hunk 4]
```diff
@@ -303,6 +303,39 @@ def delete_topics(user, user_data):
...
+        emit('edit_minute', Response.PermError)
+        return
+
+    for key in user_data:
+        if( key == "title" or key =="topic" or key == "body"):
+            setattr(minute, key, user_data[key])
+    try:
+        db.session.commit()
+        emit("edit_minute", Response.EditSuccess)
+        get_minute(user, user_data)
```
Comment (omardlhz) [inline app/minutes/controllers.py]: I don't this line is doing anything.

File: app/minutes/controllers.py [hunk 5]
```diff
@@ -303,6 +303,39 @@ def delete_topics(user, user_data):
...
+        get_minutes(user, user_data)
```
Comment (omardlhz) [inline app/minutes/controllers.py]: Same as this one.

File: app/minutes/controllers.py [hunk 6]
```diff
@@ -303,6 +303,39 @@ def delete_topics(user, user_data):
...
+    except Exception as e:
```
Comment (omardlhz) [inline app/minutes/controllers.py]: Please test exception, look at the other tests and mimic the process.

File: app/minutes/controllers.py (hunk 4 above)
Comment (moiseslorap) [reply app/minutes/controllers.py]: Which line specifically????

File: app/minutes/controllers.py (hunk 5 above)
Comment (moiseslorap) [reply app/minutes/controllers.py]: Which line specifically????

File: app/minutes/controllers.py (hunk 4 above)
Comment (moiseslorap) [reply app/minutes/controllers.py]: 334?

File: app/minutes/controllers.py (hunk 5 above)
Comment (moiseslorap) [reply app/minutes/controllers.py]: 335?

File: app/minutes/controllers.py (hunk 4 above)
Comment (omardlhz) [reply app/minutes/controllers.py]: Yes 334

File: app/minutes/controllers.py (hunk 5 above)
Comment (omardlhz) [reply app/minutes/controllers.py]: Also 335

File: app/minutes/controllers.py (hunk 6 above)
Comment (omardlhz) [reply app/minutes/controllers.py]: We'll figure out how to test this one later.

**Pull request merged**